src/games/strategy/triplea/ai/proAI/data/ProMoveOptions.java

```diff
@@ -0,0 +1,116 @@
+package games.strategy.triplea.ai.proAI.data;
+
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ProMoveOptions {
+
+  private final ProUtils utils;
+  private final ProBattleUtils battleUtils;
+  private final Map<Territory, ProTerritory> maxMoveMap;
+  private final Map<Territory, List<ProTerritory>> moveMaps;
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils) {
+    this.utils = utils;
+    this.battleUtils = battleUtils;
+    maxMoveMap = new HashMap<Territory, ProTerritory>();
+    moveMaps = new HashMap<Territory, List<ProTerritory>>();
+  }
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils,
+      final List<Map<Territory, ProTerritory>> moveMapList, final PlayerID player, final boolean isAttacker) {
+    this(utils, battleUtils);
+    populateMaxMoveMap(moveMapList, player, isAttacker);
+    populateMoveMaps(moveMapList);
+  }
+
+  public ProTerritory getMax(final Territory t) {
+    return maxMoveMap.get(t);
+  }
+
+  public List<ProTerritory> getAll(final Territory t) {
+    return moveMaps.get(t);
+  }
+
+  @Override
+  public String toString() {
+    return maxMoveMap.toString();
+  }
+
+  private void populateMaxMoveMap(final List<Map<Territory, ProTerritory>> moveMaps, final PlayerID player,
+      final boolean isAttacker) {
+
+    // Get players in turn order
+    final List<PlayerID> players = utils.getOtherPlayersInTurnOrder(player);
```

Comment:
Some more refactoring can be done to simplify. On line 56 we grab the player turn order from a `utils` object to then pass that same data back to the same `utils` object on line 72. Seems like then the call on line 56 for player turn orders is not needed, and that `isPlayersTurnFirst` on line 72 can calculate the data itself.